crates/egui/src/widgets/plot/mod.rs

```diff
@@ -799,23 +708,26 @@ impl Plot {
         let mut bounds = *last_screen_transform.bounds();
 
         // Find the cursors from other plots we need to draw
-        let draw_cursors: Vec<Cursor> = if let Some(group) = linked_cursors.as_ref() {
-            let mut frames = group.frames.borrow_mut();
+        let draw_cursors: Vec<Cursor> = if let Some((id, _)) = linked_cursors.as_ref() {
+            let data = &mut ui.memory().data;
+            let frames: &mut HashMap<Id, Vec<PlotFrameCursors>> =
```

Comment:
`HashMap<Id, Vec<PlotFrameCursors>>` is a pretty complex type that needs to match exactly when reading and writing data to `memory().data`. It would be a good idea to use a type alias or newtype for it.